### PR TITLE
Add nextjs-adapter api docs to readme #89

### PR DESCRIPTION
**Pull request by pmi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,33 +2,651 @@
...
+    variables: {
+        foo: 'bar',
+    },
+};
+
+const headers = {
+    'custom-header': 'header-value',
+};
+
+const result = fetchFromApi('http://domain:8000/graphql/api', body, contentPath, headers,);
```
[the remark this answers is not in the document]
Comment (alansemenov) [reply README.md:102]: `contentPath` is used here as an argument but not described in the table of arguments above.

File: README.md [hunk 2]
```diff
@@ -2,33 +2,651 @@
...
+### `ComponentRegistry`
+
+Registry containing definitions of all components (i.e. pages, parts, layouts, macros, etc. ). It is used in the runtime by `nextjs-adapter`
+to make component queries and render components. It has several public methods:
+
+---
+
+#### `setCommonQuery(query: SelectedQueryMaybeVariablesFunc): void`
+
+Setups the common query that is going to be executed along with component queries and passed to every component on the page.
```
Comment (alansemenov) [inline README.md:208]:
```suggestion
Sets up a common query that is going to be executed along with component queries and passed to every component on the page.
```

File: README.md [hunk 3]
```diff
@@ -2,33 +2,651 @@
...
+import {ComponentRegistry} from '@enonic/nextjs-adapter';
+
+const query = ComponentRegistry.getCommonQuery();
+```
+
+---
+
+#### `getByComponent(component: PageComponent): ComponentDefinition | undefined`
+
+Gets the definition of component stored in the ComponentRegistry by component.
```
Comment (alansemenov) [inline README.md:262]:
```suggestion
Gets component definition from the ComponentRegistry.
```

File: README.md [hunk 4]
```diff
@@ -2,33 +2,651 @@
...
+
+### `UrlProcessor`
+
+Helper singleton for processing URLs.
+
+---
+
+#### <a id="process"></a>`static process(url: string, meta: MetaData): string`
+
+Processes the absolute URL to become relative for the current viewer, while keeping in mind Next.js assets and Enonic XP binary content
```
Comment (alansemenov) [inline README.md:391]: Unnecessary linebreak?

File: README.md [hunk 5]
```diff
@@ -2,33 +2,651 @@
...
+            ${richTextQuery('htmlField')}
+        }
+    }
+}`;
 ```
 
-## Problems
+### `ComponentRegistry`
+
+Registry containing definitions of all components (i.e. pages, parts, layouts, macros, etc. ). It is used in the runtime by `nextjs-adapter`
```
Comment (alansemenov) [inline README.md:201]: Unnecessary linebreak?

File: README.md [hunk 6]
```diff
@@ -2,33 +2,651 @@
...
+import {getUrl} from '@enonic/nextjs-adapter';
+
+const urlRelativeToViewer = getUrl('http://domain:8000/graphql/api', meta);
+```
+
+---
+
+### <a id="rich-text-query"></a>`richTextQuery(fieldName: string) => string`
+
+This is a utility function for querying for `RichTextData` needed for [RichTextView](#rich-text-view). It creates a graphql query string for
```
Comment (alansemenov) [inline README.md:176]: Unnecessary linebreak?

File: README.md [hunk 7]
```diff
@@ -2,33 +2,651 @@
...
+
+---
+
+### Functions
+
+They are available at the library root at `@enonic/nextjs-adapter`
+
+### <a id="fetch-content"></a>`fetchContent(contentPath: string | string[], context: Context) => Promise<FetchContentResult>`
+
+Fetches the content/component by path with component queries optimization, page structure as well as runtime info calculation. This is the
```
Comment (alansemenov) [inline README.md:43]: Unnecessary linebreak?

File: README.md [hunk 8]
```diff
@@ -2,33 +2,651 @@
...
-Just import required functions and types in your project:
+Just import required functions and views in your project:
 
 ```tsx
-import {fetchContent} from "@enonic/nextjs-adapter/guillotine/fetchContent";
+// all functions and interfaces are accessible at the library root
+import {fetchContent} from "@enonic/nextjs-adapter";
 
-// ...
+// views are placed in a folder called views and are default exports in files
```
Comment (alansemenov) [inline README.md:23]:
```suggestion
// views are placed in a folder called `views` and are default exports in files
```

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (pmi) [reply README.md:391]: Doesn't affect the output, but easier to navigate the source

File: README.md (hunk 5 above)
Comment (pmi) [reply README.md:201]: Doesn't affect the output, but easier to navigate the source

File: README.md (hunk 6 above)
Comment (pmi) [reply README.md:176]: Doesn't affect the output, but easier to navigate the source

File: README.md (hunk 7 above)
Comment (pmi) [reply README.md:43]: Doesn't affect the output, but easier to navigate the source

**Review by alansemenov — APPROVED** [review]

**Pull request merged**